src/createAnimatedComponent.tsx

```diff
@@ -254,6 +261,27 @@ export default function createAnimatedComponent(
       }
     }
 
+    // The system is best designed when setNativeProps is implemented. It is
+    // able to avoid re-rendering and directly set the attributes that changed.
+    // However, setNativeProps can only be implemented on native components
+    // If you want to animate a composite component, you need to re-render it.
+    // In this case, we have a fallback that uses forceUpdate.
+    _animatedPropsCallback = () => {
+      if (this._component == null) {
+        // AnimatedProps is created in will-mount because it's used in render.
+        // But this callback may be invoked before mount in async mode,
+        // In which case we should defer the setNativeProps() call.
+        // React may throw away uncommitted work in async mode,
+        // So a deferred call won't always be invoked.
+        this._invokeAnimatedPropsCallbackOnMount = true;
+      } else if (typeof this._component.setNativeProps !== 'function') {
+        this.forceUpdate();
+      } else {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        this._component.setNativeProps(this._propsAnimated!.__getValue());
+      }
+    };
+
```

Comment:
Both `_invokeAnimatedPropsCallbackOnMount` and `_propsAnimated` were removed in 41956a26933da1af1c2509dd8655fefcee6657c2, I'm not sure whether this should be removed as well or not.

Reply:
To be done in #3204.